src/crypto/crypto.h

```diff
@@ -335,11 +335,19 @@ namespace crypto {
 
   inline bool operator<(const public_key &p1, const public_key &p2) { return memcmp(&p1, &p2, sizeof(public_key)) < 0; }
   inline bool operator>(const public_key &p1, const public_key &p2) { return p2 < p1; }
+  inline bool operator<(const key_image &p1, const key_image &p2) { return memcmp(&p1, &p2, sizeof(key_image)) < 0; }
+  inline bool operator>(const key_image &p1, const key_image &p2) { return p2 < p1; }
```

Comment:
Looks good all around, but just personal preference:
```suggestion
  inline bool operator>(const key_image &p1, const key_image &p2) { return memcmp(&p1, &p2, sizeof(key_image)) > 0; }
```
There's a 90% chance this will get optimized out by the compiler, but since this is gonna likely take place many, many times a second , it might be better to write it out explicitly.

Reply:
The other operator is `inline`. If this is not inlined then idk what to say - compilers must be trash.

Reply:
This comment is old, but also the `inline` keyword is needed here or linker errors could occur.